lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php

```diff
@@ -651,7 +651,11 @@ private function completeIdGeneratorMapping(ClassMetadataInfo $class): void
 
     private function determineIdGeneratorStrategy(AbstractPlatform $platform): int
     {
-        if ($platform->getName() === 'oracle' || $platform->getName() === 'postgresql') {
+        if (
+            $platform instanceof Platforms\OraclePlatform
+            || $platform instanceof Platforms\PostgreSQL94Platform
```

Comment:
Maybe we should build against DBAL `3.2@dev`, that way you can check for `PostgreSQLPlatform`, since it was reinstated in https://github.com/doctrine/dbal/pull/4755 ?

Reply:
I don't know the ETA for DBAL 3.2.0. If DBAL 3.2 makes a stable release before ORM 2.10, I'll happily bump and remove the compat code again.